x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_embeddable_factory.tsx

```diff
@@ -0,0 +1,290 @@
+ */
+export const getServices = async (
+  getStartServices: StartServicesAccessor<MlStartDependencies, MlPluginStart>
+): Promise<AnomalySwimlaneEmbeddableServices> => {
+  const [
+    [coreStart, pluginsStart],
+    { AnomalyDetectorService },
+    { AnomalyTimelineService },
+    { mlApiServicesProvider },
+    { mlResultsServiceProvider },
+  ] = await Promise.all([
+    getStartServices(),
+    import('../../application/services/anomaly_detector_service'),
+    import('../../application/services/anomaly_timeline_service'),
+    import('../../application/services/ml_api_service'),
+    import('../../application/services/results_service'),
+  ]);
+
+  const httpService = new HttpService(coreStart.http);
+  const anomalyDetectorService = new AnomalyDetectorService(httpService);
+  const anomalyTimelineService = new AnomalyTimelineService(
+    pluginsStart.data.query.timefilter.timefilter,
+    coreStart.uiSettings,
+    mlResultsServiceProvider(mlApiServicesProvider(httpService))
+  );
+
+  return [
+    coreStart,
+    pluginsStart as MlDependencies,
+    { anomalyDetectorService, anomalyTimelineService },
+  ];
+};
+
+export const getAnomalySwimLaneEmbeddableFactory = (
+  getStartServices: StartServicesAccessor<MlStartDependencies, MlPluginStart>
+) => {
+  const factory: ReactEmbeddableFactory<
+    AnomalySwimLaneEmbeddableState,
+    AnomalySwimLaneEmbeddableApi
+  > = {
+    type: ANOMALY_SWIMLANE_EMBEDDABLE_TYPE,
+    deserializeState: (state) => {
+      return state.rawState as AnomalySwimLaneEmbeddableState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const services = await getServices(getStartServices);
+
+      const [coreStartServices, pluginsStartServices, anomalySwimLaneServices] = services;
+
+      const subscriptions = new Subscription();
+
+      const interval = new BehaviorSubject<number | undefined>(undefined);
+
+      const dataLoading = new BehaviorSubject<boolean | undefined>(true);
+      const blockingError = new BehaviorSubject<Error | undefined>(undefined);
+      const query$ =
+        // @ts-ignore
+        (state.query ? new BehaviorSubject(state.query) : parentApi?.query$) ??
+        new BehaviorSubject(undefined);
+      const filters$ =
+        // @ts-ignore
+        (state.query ? new BehaviorSubject(state.filters) : parentApi?.filters$) ??
+        new BehaviorSubject(undefined);
+
+      const refresh$ = new BehaviorSubject<void>(undefined);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const { appliedTimeRange$, serializeTimeRange, timeRangeComparators, timeRangeApi } =
+        initializeTimeRange(state, parentApi);
+
+      const { swimLaneControlsApi, serializeSwimLaneState, swimLaneComparators } =
+        initializeSwimLaneControls(state, titlesApi);
+
+      // Helpers for swim lane data fetching
+      const chartWidth$ = new BehaviorSubject<number | undefined>(undefined);
```

Comment:
I'm curious about why the chart width changing causes the data to re-query. Is it possible to only query when something changes, and have the rendered component just re-draw itself with a different width?